src/gauge.js

```diff
@@ -261,75 +275,196 @@
+        if (!elements.length) return false;
+        do {
+          gaugeElement.removeChild(elements[0]);
+        } while (elements.length);
+      }
+
+      function setGaugeColor(value, duration) {
+        var c = gaugeColor(value),
+          dur = duration * 1000,
+          pathTransition = "stroke " + dur + "ms ease";
+        // textTransition = "fill " + dur + "ms ease";
 
         gaugeValuePath.style = [
           "stroke: " + c,
           "-webkit-transition: " + pathTransition,
           "-moz-transition: " + pathTransition,
           "transition: " + pathTransition,
         ].join(";");
-        /*
-        gaugeValueElem.style = [
-          "fill: " + c,
-          "-webkit-transition: " + textTransition,
-          "-moz-transition: " + textTransition,
-          "transition: " + textTransition,
-        ].join(";");
-        */
       }
 
       instance = {
         setMaxValue: function(max) {
           limit = max;
         },
-        setValue: function(val) {
-          value = normalize(val, min, limit);
-          if(gaugeColor) {
-            setGaugeColor(value, 0)
-          }
-          updateGauge(value);
+        setValue: function (val) {
+          value = normalizeMultipleValue(val, min, limit, gaugeColor);
+          updateGauge(value, 0, 0);
         },
-        setValueAnimated: function(val, duration) {
-        	var oldVal = value;
-          value = normalize(val, min, limit);
-          if(oldVal === value) {
+        setValueAnimated: function (val, duration) {
+          clearGaugeSections();
+          duration = (duration / val.length).toFixed(2) || 1;
+          var oldVal = value;
+          value = normalizeMultipleValue(val, min, limit, opts.color);
+          if (oldVal === value) {
             return;
           }
 
-          if(gaugeColor) {
+          if (gaugeColor) {
             setGaugeColor(value, duration);
           }
-          Animation({
-            start: oldVal || 0,
-            end: value,
-            duration: duration || 1,
-            step: function(val, frame) {
-              updateGauge(val, frame);
-            }
+
+          // This sum of section end point need for set valid start point for each sections
+          var totalAngle = 0,
+            label;
+
+
+          value.forEach(function (v, i) {
+            // Set startSectionPoint for each section
+            value[i].startSectionPoint = totalAngle;
+
+            label = generateMultipleLabel(label, v.value);
+
+            // Write section step by step by duration
+            setTimeout(function () {
+              Animation({
+                start: 0,
+                end: v.value,
+                duration,
+                step: function (val, frame) {
+                  updateSingleGaugeSection(
+                    {value: val, color: v.color},
+                    value[i].startSectionPoint
+                  );
+                }
+              });
+            }, duration * 1000 * i);
```

Comment:
why do we need to multiply duration by 1000?

Reply:
duration in secconds